hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -2109,60 +2109,60 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
      * <p>
      * The {@code EntryProcessor} may implement the {@link Offloadable} and {@link ReadOnly} interfaces.
      * <p>
-     * If the EntryProcessor implements the {@link Offloadable} interface the processing will be offloaded to the given
-     * ExecutorService allowing unblocking of the partition-thread, which means that other partition-operations
+     * If the EntryProcessor implements the {@link Offloadable} interface, the processing will be offloaded to the given
+     * ExecutorService, allowing unblocking of the partition-thread, which means that other partition-operations
      * may proceed. The key will be locked for the time-span of the processing in order to not generate a write-conflict.
      * In this case the threading looks as follows:
      * <ol>
      * <li>partition-thread (fetch & lock)</li>
      * <li>execution-thread (process)</li>
      * <li>partition-thread (set & unlock, or just unlock if no changes)</li>
      * </ol>
-     * If the EntryProcessor implements the Offloadable and ReadOnly interfaces the processing will be offloaded to the
+     * If the EntryProcessor implements the Offloadable and ReadOnly interfaces, the processing will be offloaded to the
      * given ExecutorService allowing unblocking the partition-thread. Since the EntryProcessor is not supposed to do
-     * any changes to the Entry the key will NOT be locked for the time-span of the processing. In this case the threading
+     * any changes to the Entry, the key will NOT be locked for the time-span of the processing. In this case the threading
      * looks as follows:
      * <ol>
-     * <li>partition-thread (fetch & lock)</li>
+     * <li>partition-thread (fetch)</li>
```

Comment:
WHat happened to the &lock?

Reply:
I think it's not supposed to be there. On line 2123 it says "the key will NOT be locked" (emphasis was not added), also there's no unlock. I didn't check the actual implementation though.

Reply:
Direct message from Ahmet Mırçık Today at 11:36
>Hi, your change is correct, there is no locking for read-only entry processors